1_fetch_spatial.R

```diff
@@ -0,0 +1,187 @@
+    p1_lakes_sf,
+    {read_csv('1_fetch/in/saline_lakes.csv', col_types = 'ccnn') %>% 
+        st_as_sf(coords = c('Lon','Lat'), crs = selected_crs) %>% 
+        rename(point_geometry = geometry, lake = `Lake Ecosystem`, state = State) %>% 
+        mutate(state_abbr = case_when(state == 'California' ~ 'CA',
+                                      state == 'Nevada' ~ 'NV',
+                                      state == 'Utah' ~ 'UT',
+                                      state == 'Oregon' ~ 'OR',
+                                      state == 'California/Oregon' ~ 'CA', # Put lake crossing OR CA border as CA for now.  
+                                      TRUE ~ 'NA'),
+               lake = str_to_title(lake),
+               lake_w_state = paste(lake, state_abbr, sep = ','),
+               lake_name_shrt = trimws(str_replace(lake, pattern = 'Lake', replacement = "")))
+    }
+  ),
+  
+  # States Shp Fetch - used for Lakes Querying #
+  
+  ## Download states shp
+  tar_target(
+    p1_download_states_shp,
+    download_states_shp(url = states_download_url, 
+                        out_path = '1_fetch/in/states_shp'),
+    format = 'file'
+  ),
+  
+  tar_target(
+    p1_states_sf,
+    st_read(file.path(p1_download_states_shp,'statesp010g.shp'), quiet = TRUE) %>%
+      filter(STATE_ABBR %in% c('CA',"NV",'UT','OR')) %>% 
+      st_transform(crs = st_crs(p1_lakes_sf)) %>% 
+      select(NAME,STATE_ABBR, geometry)
+  ),
+  
+  
+  # nhdhr download and fetch #
+  
+  ## 1st fetch of huc08  to get all relevant high res nhd data (water bodies, huc6, huc8, huc10 areas) for focal lakes
+  tar_target(
+    p1_huc08_full_basin_sf,
+    get_huc8(AOI = p1_lakes_sf$point_geometry)
+  ),
+  
+  ## Split huc ids to 04 to pull nhdhr - download_nhdhr() requires that huc id param be huc4.
+  tar_target(
+    p1_huc04_for_download,
+    substr(p1_huc08_full_basin_sf$huc8, start = 1, stop = 4) %>%
+        unique() %>%
+        ## adding 1601 which is a HU4 that contains watersheds relevant to Great Salt Lake 
+        append('1601')
+  ),
+  
+  ## Download high res nhd data to get lake water bodies 
+  tar_target(
+    p1_download_nhdhr_lakes_path,
+    download_nhdhr_data(nhdhr_gdb_path = '1_fetch/in/nhdhr',
+                        huc04_list = p1_huc04_for_download),
+    format = 'file'
+  ),
+  
+  ## Fetch waterbodies, huc6, huc8, huc10 from hr and place in local gpkg
+  tar_target(p1_nhd_gpkg, 
+             get_downloaded_nhd_data(gdb_path = p1_download_nhdhr_lakes_path,
+                                     out_gpkg_path = '1_fetch/in/nhd_WB_HUC6_HU8_HU10.gpkg',
+                                     layer = c('NHDWaterbody','WBDHU6', 'WBDHU8', 'WBDHU10')),
+             format = 'file'
+  ),
+  
+  # Lakes Fetch #
+  
+  ## Read in all nhd hr waterbodies in our entire basin
+  ## This is then processed to our focal lakes in 2_process.R
+  tar_target(p1_nhdhr_lakes,
+             sf::st_read('1_fetch/in/nhd_WB_HUC6_HU8_HU10.gpkg',
```

Comment:
Here I needed to provide the target name `p1_nhd_gpkg` that corresponds to the `1_fetch/in/nhd_WB_HUC6_HU8_HU10.gpkg` file path, because otherwise it was building different targets that depend on it first and failing.

Reply:
But once that it built locally, it makes sense to use the file path like you're doing to avoid the lengthy `skip` steps when running the pipeline